### DISCUSSION TITLE
同步服务器密码或文件包含关键字异常（@，:）

### DISCUSSION THREAD

**Discussion by malin177** [juicedata/juicefs#2962] [General]
**What happened**:

 juicefs sync root:{@1234:12313}@xx.xx.xx.xx:22:/root/j@fs s3://xxxx:xxx@xxx.xxx.xxx.xxx:80/test/jfs

**Anything else we need to know?**

版本：1.0.0-rc3
sftp异常代码位置：cmd/sync.go>223行
参考处理方式：
![image](https://user-images.githubusercontent.com/35835802/200793265-8ac32dc0-3335-4ade-b24c-8d5dde681de3.png)

ps: 其他存储可能存在类似问题
建议多点debug日志，异常发生比较难定位

**Comment by zhijian-pro** [comment]
密码如果包含了特殊字符，则需要手动转义并替换，比如 / 需要被替换为其转义符 %2F
把@与：转一下再试试
https://juicefs.com/docs/zh/community/guide/sync/#%E5%9F%BA%E6%9C%AC%E7%94%A8%E6%B3%95

**Reply by malin177** [reply]
密码使用urlencode编码也会异常，代码中密码并没有进行解码，下面是异常日志：
2022/11/14 09:56:43.076443 juicefs[109181] <ERROR>: connect to 10.201.1.17 failed: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none password publickey], no supported methods remain [sftp.go:539]
2022/11/14 09:56:43.076567 juicefs[109181] <FATAL>: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none password publickey], no supported methods remain [main.go:31]

**Comment by zhijian-pro** [comment]
的确是个 bug，我们在这里进行修复了https://github.com/juicedata/juicefs/pull/2981
